File: 02a.js
import * as fs from 'fs';

const filename = "02.dat";
fs.readFile(filename, (err, data) => {
    if (err) throw err;

    var x = 0;
    var y = 0;

    data.toString().split("\n").forEach(command => {
        var op = command.split(" ");
		if (op[0] === "forward") {
            x += parseInt(op[1]);
        }
        else if (op[0] === "down") {
            y += parseInt(op[1]);
        }
        else if (op[0] === "up") {
            y -= parseInt(op[1]);
        }
	});

    console.log(x*y);
});

File: 02a.py
#!/usr/bin/python3

with open('02.dat') as f:
    data = f.read()
commands = data.split('\n')
commands.pop()

x = 0
y = 0
for command in commands:
    op = command.split(' ')
    if (op[0] == 'forward'):
        x += int(op[1])
    elif (op[0] == 'down'):
        y += int(op[1])
    elif (op[0] == 'up'):
        y -= int(op[1])
print(x*y)